activemodel/lib/active_model/secure_password.rb

```diff
@@ -93,15 +93,17 @@ def has_secure_password(attribute = :password, validations: true)
         if validations
           include ActiveModel::Validations
 
+          validation_options = validations.is_a?(Hash) ? validations : {}
+
           # This ensures the model has a password by checking whether the password_digest
           # is present, so that this works with both new and existing records. However,
           # when there is an error, the message is added to the password attribute instead
           # so that the error message will make sense to the end-user.
-          validate do |record|
+          validate(validation_options) do |record|
             record.errors.add(attribute, :blank) unless record.public_send("#{attribute}_digest").present?
           end
 
-          validate do |record|
+          validate(validation_options) do |record|
```

Comment:
I am not sure this is the behaviour I would expect though. It could still hold true that you set a `password_challenge` but there might still be situation where you don't want to validate it. If I was to use this library, I would expect that doing:

`validations { if: :my_condition }`

 would disable _all_ the validations, not _some_ of them.

Reply:
I understand that those are just params that will be set. So you wouldn't be setting those params if you wanted to skip validations, so I get what you are saying, but feels like it would still hold true that my expectation would be to disable all validations regardless (just in case you are still sending those params for some reasons). It's more in term of what the API looks like.

Reply:
`password_challenge` is a gating mechanism that checks the *already existing* password when e.g. changing some sensitive field of the model.  It is a temporary value, like `password_confirmation`.  See #43688 for more information.

If a client sends a `password_challenge` we should definitely check it.  If a client sends a `password_challenge` unintentionally, I would consider that an error (which the validation failure would then reflect).  And if an application needs different behavior, it can still use `validations: false` as an escape hatch.

Reply:
Yes, that makes sense. I agree it should be an error. Not sure what I was thinking, Probably poorly understand what the password_challenge was and didn't really look at it TBH. Thanks for that